frontend/src/app/tasks/hooks/useCreateTaskForm.ts

```diff
@@ -54,19 +56,31 @@ export const useTaskForm = () => {
     addTask(formState)
   }, [addTask, formState, showError, tasks])
 
-  const setDate = () => handleChange('date', format(new Date(), 'yyyy-MM-dd'))
+  const selectTemplate = (templateId: number, templates: Array<Template>) => {
+    const template = templates.find((t) => t.id === templateId)
+    if (template) {
+      setTemplateName(template.name)
+      setFormState((prevState) => ({
+        ...prevState,
+        taskType: template.taskType,
+        description: template.description || '',
+        startTime: template.startTime,
+        endTime: template.endTime,
+        projectId: template.projectId.toString(),
```

Comment:
I noted it elsewhere in the review, but the projectId in a template is nullable, so this will throw an error in the case it is null. I also wondered why we are using toString() on the projectId?

Reply:
hmmm, it should be a number actually, I will fix it